components/more_direct_channels/more_direct_channels.jsx

```diff
@@ -282,7 +301,7 @@ export default class MoreDirectChannels extends React.Component {
     }
 
     handleSubmitImmediatelyOn = (value) => {
-        return value.id === this.props.currentUserId || value.delete_at;
+        return value.id === this.props.currentUserId || Boolean(value.delete_at);
```

Comment:
Not related to the issue but decided to fix while I'm adding test to this component.  The consumer of this function uses it for boolean flags and so returning an explicit boolean value is the best to do it.